### PR TITLE
Make shaving/haircut interruptable

### PR DESCRIPTION
**Pull request by lucasmr**
Fixes #24295. Moves haircut and shaving to new activity handlers. This way, these actions can be interrupted by approaching enemies.

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -7572,18 +7572,14 @@ int iuse::shavekit( player *p, item *it, bool, const tripoint & )
     if( !it->ammo_sufficient() ) {
         p->add_msg_if_player( _( "You need soap to use this." ) );
     } else {
-        p->add_msg_if_player( _( "You open up your kit and shave." ) );
-        p->moves -= 3000;
-        p->add_morale( MORALE_SHAVE, 8, 8, 240_minutes, 3_minutes );
+        p->assign_activity( activity_id( "ACT_SHAVE" ), 3000);
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]: Whitespace is missing after 3000.

File: src/iuse.cpp [hunk 2]
```diff
@@ -7572,18 +7572,14 @@ int iuse::shavekit( player *p, item *it, bool, const tripoint & )
...
     }
     return it->type->charges_to_use();
 }
 
 int iuse::hairkit( player *p, item *it, bool, const tripoint & )
 {
-    p->add_msg_if_player( _( "You give your hair a trim." ) );
-    p->moves -= 3000;
-    p->add_morale( MORALE_HAIRCUT, 3, 3, 480_minutes, 3_minutes );
+    p->assign_activity( activity_id( "ACT_HAIRCUT" ), 3000);
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]: Missing whitespace after 3000 here too.

File: data/json/player_activities.json [hunk 3]
```diff
@@ -343,5 +343,17 @@
...
+    "type": "activity_type",
+    "stop_phrase": "Stop shaving?",
+    "based_on": "speed"
+  },
+  {
+    "id": "ACT_HAIRCUT",
+    "type": "activity_type",
+    "stop_phrase": "Stop cutting your hair?",
+    "based_on": "speed"
+   }
```
Comment (Night-Pryanik) [inline data/json/player_activities.json]: There should be 2 spaces instead of 3.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/cs-students-looking-to-fix-issue-23440/17737/4

**Pull request merged**